### PR TITLE
Unit cell volumes

### PR DESCRIPTION
**Pull request by rlavendomme**
Added function to count number of voxels per type within the unit cell space.
Modified surface map export function to export the surface map within the unit cell only.

### CODE REVIEW

File: src/controller.cpp [hunk 1]
```diff
@@ -105,13 +105,13 @@ bool Ctrl::runCalculation(){
...
     notifyUser("Result for " + data.chemical_formula);
     notifyUser("Elapsed time: " + std::to_string(data.getTime()) + " s");
-    notifyUser("Van der Waals volume: " + std::to_string(data.volumes[0b00000011]) + " A^3");
-    notifyUser("Excluded void volume: " + std::to_string(data.volumes[0b00000101]) + " A^3");
-    notifyUser("Probe 1 core volume: " + std::to_string(data.volumes[0b00001001]) + " A^3");
-    notifyUser("Probe 1 shell volume: " + std::to_string(data.volumes[0b00010001]) + " A^3");
+    notifyUser("Van der Waals volume: " + std::to_string(data.volumes[0b00000011]) + " �^3");
+    notifyUser("Excluded void volume: " + std::to_string(data.volumes[0b00000101]) + " �^3");
+    notifyUser("Probe 1 core volume: " + std::to_string(data.volumes[0b00001001]) + " �^3");
+    notifyUser("Probe 1 shell volume: " + std::to_string(data.volumes[0b00010001]) + " �^3");
```
Comment (jmaglic) [inline src/controller.cpp:111]: None of these lines are output on macOS. I assume it's because of the Å

File: src/controller.cpp [hunk 2]
```diff
@@ -105,13 +105,13 @@ bool Ctrl::runCalculation(){
...
     if(data.probe_mode){
-      notifyUser("Probe 2 core volume: " + std::to_string(data.volumes[0b00100001]) + " A^3");
-      notifyUser("Probe 2 shell volume: " + std::to_string(data.volumes[0b01000001]) + " A^3");
+      notifyUser("Probe 2 core volume: " + std::to_string(data.volumes[0b00100001]) + " �^3");
+      notifyUser("Probe 2 shell volume: " + std::to_string(data.volumes[0b01000001]) + " �^3");
```
Comment (jmaglic) [inline src/controller.cpp:114]: See above

**Review by jmaglic — CHANGES_REQUESTED** [review]
I noticed an issue: After I select isobutane.pdb for calculation and check the "Analyse crystal unit cell" option I get an error "Space group or symmetry not found!". Afterwards I cannot start any new calculation, neither by unticking the checkbox nor by selecting a new file. Instead I receive "Calculation failed!" every time I press "Calculate".

Also how do I get the unit cell tick box to work?

File: src/controller.cpp (hunk 1 above)
Comment (rlavendomme) [reply src/controller.cpp:111]: Interesting. Does the report file output properly?

**Comment by rlavendomme** [comment]
> I noticed an issue: After I select isobutane.pdb for calculation and check the "Analyse crystal unit cell" option I get an error "Space group or symmetry not found!". Afterwards I cannot start any new calculation, neither by unticking the checkbox nor by selecting a new file. Instead I receive "Calculation failed!" every time I press "Calculate".

Nice catch. Originally when the unit cell processing functions were introduced, this kind of error would have sent back the gui to normal. I need to fix this in this build.
 
> Also how do I get the unit cell tick box to work?

You simply need to use a pdb file that contains crystallographic information, which is not the case of isobutane.pdb in the inputfile folder. I'll add some pdb files with crystallographic information.

File: src/controller.cpp (hunk 1 above)
Comment (jmaglic) [reply src/controller.cpp]: Yes that works fine

**Review by jmaglic — COMMENTED** [review]
The previous issue has been solved, but with either test_unit_cell_1 and test_unit_cell_2 I'm getting

> Space group or symmetry not found!
Check the structure and space group files
or untick the unit cell analysis checkbox.
Calculation failed!

 Is this only on my platform? Am I missing some file?

**Comment by rlavendomme** [comment]
> 
> 
> The previous issue has been solved, but with either test_unit_cell_1 and test_unit_cell_2 I'm getting
> 
> > Space group or symmetry not found!
> > Check the structure and space group files
> > or untick the unit cell analysis checkbox.
> > Calculation failed!
> 
> Is this only on my platform? Am I missing some file?

Do you have space_groups.txt in the ./inputfile folder ?

**Pull request merged**